Change pending cluster state to handle multiple bucket spaces

toregge opened the pull request:
@geirst, @vekterli: please review

Commit:
Pass distributor bucket space repo to pending cluster state.

Commit:
Add test method to get a specific pending bucket space db transition from a
pending cluster state.  Remove existing test methods that forwarded
to the pending bucket space db transition.

Commit:
Rename mergeInto() to MergeIntoBucketDatabases() and remove argument,
to remove limitation of a single bucket database.

Commit:
Change pending cluster state to handle more than one bucket space.

geirst commented on storage/src/vespa/storage/distributor/pendingclusterstate.cpp:
```diff
@@ -64,19 +64,26 @@ PendingClusterState::PendingClusterState(
+PendingClusterState::~PendingClusterState() {}
+
+void
+PendingClusterState::constructorHelper()
```
Consider rename to something like "considerRequestBucketInfo"

geirst replied:
Or "init pending transitions"

vekterli commented on storage/src/vespa/storage/distributor/pendingclusterstate.cpp:
```diff
@@ -64,19 +65,28 @@ PendingClusterState::PendingClusterState(
     logConstructionInformation();
     markAllAvailableNodesAsRequiringRequest();
-    _pendingTransition = std::make_unique<PendingBucketSpaceDbTransition>(*this, _clusterInfo, _newClusterState, _creationTimestamp);
+    constructorHelper();
```
I'd prefer a more descriptive name here. Maybe `initializeBucketSpaceTransitions` or `sendInitialBucketSpaceRequests` ?

vekterli reviewed (approved):
:+1: see naming and logging nitpicks

geirst reviewed (approved):
LGTM. Please consider my comments

Commit:
Improve method naming: constructorHelper -> initializeBucketSpaceTransitions.
Print bucket space in debug log message when requesting bucket info for node.
Adjust comment for setNodeReplied method.

geirst reviewed (approved):
👍
